src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -185,7 +187,9 @@ void assert_lock_strong(const Mutex* lock) {
 
 static void add_mutex(Mutex* var) {
   assert(_num_mutex < MAX_NUM_MUTEX, "increase MAX_NUM_MUTEX");
-  _mutex_array[_num_mutex++] = var;
+  int id = _num_mutex++;
+  _mutex_array[id] = var;
+  var->set_id(id);
```

Comment:
It is not obvious how the setting of the `_id` by the constructor interacts with this code. So IIUC the `_id` of each Mutex in the array is the Mutex's index into the array. So what does `_id` mean of mutexes that are not in the array?

Reply:
This change is actually incorrect and unnecessary. The `name2Id()` function will set the Mutex's `_id` which will be used as an index into the `_names` and `_is_unique` array.